tests/integration/test_balancer.py

```diff
@@ -277,14 +280,17 @@ async def test_balancer_prepare_unit_removal(self, ops_test: OpsTest):
 
         assert balancer_is_ready(ops_test=ops_test, app_name=self.balancer_app)
 
-        rebalance_action = await leader_unit.run_action(
-            "rebalance",
-            mode="remove",
-            brokerid=new_broker_id,
-            dryrun=False,
-        )
-        response = await rebalance_action.wait()
-        assert not response.results.get("error", "")
+        for attempt in Retrying(stop=stop_after_attempt(5), wait=wait_fixed(60), reraise=True):
+            with attempt:
+                rebalance_action = await leader_unit.run_action(
```

Comment:
**question** Why is this retry needed?

Reply:
Not 100% sure, but we found that when trying to `remove` rebalance, it would say that it didn't have enough valid partitions, which definitely wasn't true. As it's an action, we don't need to guarantee that it works every time, only that it works when it's supposed to work, so I thought the retry here was appropriate.